tests/ui/typeck/method-chain-gats.stderr

```diff
@@ -0,0 +1,30 @@
+error[E0277]: the trait bound `<T as Base>::Base<C>: Functor<C>` is not satisfied
+  --> $DIR/method-chain-gats.rs:18:20
+   |
+LL |     input.fmap(f1).fmap(f2)
+   |                    ^^^^ the trait `Functor<C>` is not implemented for `<T as Base>::Base<C>`
+   |
+note: the method call chain might not have had the expected associated types
+  --> $DIR/method-chain-gats.rs:13:29
+   |
+LL | fn fmap2<T, A, B, C>(input: T, f1: impl Fn(A) -> B, f2: impl Fn(B) -> C) -> T::Base<C>
+   |                             ^ `Base::Base` is `<T as Base>::Base<_>` here
```

Comment:
wrong location is suppose?

Reply:
I think the location is "right", because the logic is looking for the root element of the method chain (even accounting for `let` bindings and `fn` arguments), so this is trying to say that `<Base>::Base<_> = <T as Base>::Base<_>`, which *is* somewhat useless information.

I do have to say that the test code makes my head spin, even without accounting for the error message 😬

Reply:
I think the only thing left to do is to improve this label to better account for GATs, but that shouldn't happen in this PR.